### DISCUSSION TITLE
pin D4 для модуля DFPlayer

### DISCUSSION THREAD

**Discussion by chipm64** [vvip-68/LedPanelWiFi#104] [Вопросы и ответы]
Почему-то не работает SoftwareSerial RX D4 esp8266. Для отладки использовал пример из библиотеки DFPlayer Mini mp3 by Makuna v1.1.1  Плеер начинает работать только, если переназначить D4 на любой другой пин. Даже переключение внутреннего светодиода с GPIO2 на GPIO16 не помогает. Но в проекте назначен D4 и пинов больше не остается при максимальной конфигурации.

**Comment by vvip-68** [comment]
Что значит "переключение", когда он в этой версии платы припаян физически на GPIO2 - то есть на D4?
Вы его перепаивали что ли? А не работает, видимо из за сочетаний каких-то физических характеристик конкретной платы и плеера. Может его физически подтягивать нужно к земле или к питанию... В моих экспериментах работал, но видимо повезло...

А поменять в максимальной конфигурации с другим пином каким-нибудь?
Тут ребята проводили исследование по назначению портов - может поможет чем...
https://github.com/vvip-68/LedPanelWiFi/wiki/Продвинутый-уровень.-Использование-портов-микроконтроллера.

Может для максимальной конфигурации разумнее было бы какой-нибудь ESP32 взять? Там портов гораздо больше...

**Reply by chipm64** [reply]
Почитал ссылку - познавательно, но к сожалению , не более
SoftwareSerial secondarySerial(0, 2); // RX, TX  не работает
SoftwareSerial secondarySerial(4, 5); // RX, TX   работает
SoftwareSerial secondarySerial(0, 5); // RX, TX работает
SoftwareSerial secondarySerial(0, 14); // RX, TX работает

Хоть тресни, не хочет GPIO2 (D4) ни под каким предлогом работать, пробовал библиотеку DFRobot - тоже самое.
Можно ли достоверно утверждать, что если не работает на отладке в сочетании (0,2) [D3 D4] то не будет работать и  в коде проекта? В проекте не предпринималось никаких мер борьбы с занятием D4 под нужды системы?
Спасибо

Понятно что в такой ситуации решением мог бы быть ESP32 да модулей ESP8266 набрался и надо их задействовать))
Может от карты памяти отказаться, наличие звуков интереснее выглядит. Или может от кнопки, не знаю пока что она делат, насколько актуально.

**Reply by vvip-68** [reply]
Используйте рабочие варианты, которые у вас определились экспериментальным путем. 
На моих тестовых устройствах, когда все это проектировалось - D4 с модулем DF-player работало без проблем, поэтому такие назначения и остались.  Единственное серьезное ограничение - SD карта должна использовать D5,D6,D7 потому что это аппаратная шина SPI и драйвер карты написан именно для использования этих пинов.

По умолчанию в проекте пин D8 назначен так же для управления SD-картой - просто потому что рядом с другими пинами и к тому же SD-card shield также использует его. Человек, составивший ту познавательную таблицу допустимости использования пинов проделал большую работу, чтобы определить какие пины с чем можно  использовать. Впрочем, кажется, высокий уровень на D8 в момент старта микроконтроллера не даст ему запуститься. Надо с этим быть осторожным.

Схема соединения для всех устройств и назначения на пины, в том числе и DF-плеер - определена им, и очевидно, у него эта схема работала. Почему не работает у вас - не знаю. Я полные наборы не собирал, нео подключение D3/D4 для плеера у меня всегда работало без проблем.

Кнопка нужна для переключения режимов вручную с кнопки, уменьшения/увеличения яркости. В принципе это все можно сделать и из приложения / веб-интерфейса. Единственный момент, чего не выполнишь из Web-интерфейса - при запуске микроконтроллера если настройка подключения к сети неправильная - он будет ожидать появления сети 3 минуты. И нажатие кнопки может прервать это ожидание. Что, в общем то можно пережить и без кнопки, поскольку такая ситуация встречается не часто.

Да, если это не работает в тестовых вариантах - это 100%не будет работать и в коде проекта.

Вообще видел всего пару человек, которые собирают устройство задействовав все подключения по максимуму. Вы, видимо, третий. Обычно это никому не нужно и используют только кнопку, матрицу и возможно расширение SD-карты.

**Reply by chipm64** [reply]
Ура! выяснилось. Не хочет работать в паре D3D4. В разных сочетаниях отдельно работает D3 и D4. А вместе - не дружат. Поменял D3 и D1  и уже в Panel заработало! Осталось не до конца понятным по папкам звуков, хоть и прочитал ваши комментарии.
Из web управляются звуки природы из папки/02/
В папку /01/ положил звуки часов  и минут. Когда они будут звучать - не понимаю, как из возбудить? И в папку /03/ звуки бегущей строки - не нашел их....
В общем помогли, спасибо. У меня часто так: начинаешь спрашивать и вдруг нахожу решение.

**Reply by vvip-68** [reply]
Кто вам сказал, что там вообще есть звуки часов и минут? Соответственно звучать они будут - никогда.
Папки - звуки будильника, рассвета, и для макроса бегущей строки {A} - включение звука привлечения внимания когда бежит строка. На этом всё.

**Reply by chipm64** [reply]
![image](https://github.com/user-attachments/assets/49983652-d752-44d0-9abf-6051b53be5a6)
![image](https://github.com/user-attachments/assets/e76322f8-cd06-4265-a5be-e0f6aa51b8b0)
Даже на мове есть) на беларускай
![image](https://github.com/user-attachments/assets/b109ac7a-eee1-40a2-91ad-dc95819319bb)

**Reply by vvip-68** [reply]
О, там в файловом хранилище вообще много чего есть, что к данному проекту не относится.
Разве где-то в описании проекта есть упоминание, что он поддерживает говорящие часы и инструкции как это все настроить?

Когда-то планы были, но изучив вопрос пришел к выводу что это никому не нужно, и даже если кто-то ставит себе говорящие часы (другие проекты) отключает их через день-два и больше не использует.

**Reply by chipm64** [reply]
Согласен полностью . 
А проект ваш интересен как раз своим набором, а не частью... Ибо, уж не обижайтесь, насчет эффектов WLED заметно превосходит. Ощущения - не хватает мощи эффектов - это конечно 70% смысла такого проекта. 
Делаю корпус, надеюсь довести до кондиции...
P.S. Сломался звук(( надо разбираться в чем дело.. что-то цвыкает.. не играет в общем.. фиг знает что... Уже ведь играло.

**Reply by vvip-68** [reply]
Проект в сеть попал вообще довольно случайно. Потому что в первую очередь делался для себя - нужна была гирлянда и прародитель от гайвера (MatrixBT) вполне под это подходил. И нужны были часы, чтобы постоянно стояли на столе и показывали время. А потом и погоду. И у Гайвера - bluetooth, а я то постоянно в сети WiFi. Пожтому - делаем WiFi. Вообще все изменения были по большей части утилитарными. Нужно было удаленно следить и управлять гирляндой - добавили MQTT. Перестало быть нужным - зато показалось удобнее управлять в браузере, а не в убогом приложении  - сделали Web-версию. И убрали MQTT ибо сделано нестандартно, а как правильно - никто не знает, и я в том числе. Частично - на "а смогу ли я " - отсюда коллекции строк, чего у других вроде не было. И прочие "фишки" типа групповой синхронизации, нестандартные кусочные матрицы, индексные файлы для их мапирования - оттуда же - интересно попробовать - смогу ли.

А проект как в самом начале утек в сеть - Гайвер у себя ссылку сделал - так не убирать же. Тем более кто-то повторяет зачем-то.
Часть эффектов от Гайвера. Часть самописанных. Часть - у кого-то видел, утянул. WLED крут (не помню есть ли исходники эффектов) - но это ж надо время выкроить, сорганизоваться, посмотреть как работает.  В общем потратить кучу времени с непонятным результатом - а нужно ли оно мне? И, как правило во всех этих проектах ламп / гирлянд - 80% эффектов на мой вкус - трэш и/или специфичен именно в варианте лампы через матовое стекло.

Вот просили "мотыльки в банке". Ну так оно только в панке и смотрится. А в виде плоской панели / гирлянды / часах - вообще не впечатляет.

А не знаю зачем все это написал. Просто сказать, что на новые другие эффекты нужно потратить много сил, на что нет то времени, то энтузиазма во все это ввязываться. Может как-нибудь будет скучно - поковыряюсь.

**Reply by chipm64** [reply]
Спасибо за обратную связь. Все прояснилось, предельно понятно и многое обьясняет. 
Кажется не сильно вас напряг, рад общению. 
Ибо у самого на увлечение не очень много времени, а отказать себе в удовольствии поковыряться в "радиодетальках"... ну как откажешь?) Внутренний позыв из детства) Посмотрим до чего доведется у меня. .... [и чего он гад этот звук стал цвыркать...]
"Тем более кто-то повторяет зачем-то."  - это мне больше всего понравилось, мой друг.. особенно))) В этом "зачем-то" много чего кроется... много

**Reply by vvip-68** [reply]
По цвырканию - проверьте напряжение питания. По даташиту DFPlayer принимает от 3.3 до 5.0 вольт, стандартное заявленное его питание - 4.2 вольта. В системах где гирлянда вот этот плеер довольно часто питают от источника в котором 5.5 вольт. При этих условиях DFPlayer вполне уже может отказываться работать. Лучше соблюдать ограничение по питанию системы в 4.8 вольта. Так и лента работает и плеер.

**Reply by chipm64** [reply]
да, спасибо, разобрался. 
Вернулся к "чистый плеер без ничего Makuna", тут и выяснилось - подвели проводки китайские dupont, раздолбались гнездовые от тыканий по пинам в экспериментах. Пальцами прижал - все ОК. Лучше конечно паять, посмотрим как останется по итогу. Можно на новые PLS планки заменить, глядишь и будет работать )
Плеер пока запитан 3.3 , линии связи - без резисторов, динамик 8 ом, так что большого тока потребления от него не ожидается (позже измерю), а вот лента на 5.1-5.2 прибл. Все в стадии макетирования. Вот когда в корпус вкину, тогда все окончательно перепроверю с учетом вами сказанного и наблюдения за поведением.
Надо еще разобраться с картой памяти , какие из нее эффекты и чтобы вообще заработала.. оставлял на последнюю очередь.

Про переключение светодиода на GPIO16. Это я так думал. В ардуино ИДЕ есть т.н. выбор светодиода 2/16 Вот я и подумал, что на стадии компилирования есть возможность скоммутировать его на другую ногу (ожидая, что это поддерживается схемотехникой), оказалось, что никакого эффекта это не произвело. Наверное для особых версий плат. Схему NodeMCU нашел - в ней вообще подключен именно к 16. У меня по факту - на 2.

**Reply by vvip-68** [reply]
Имейте в виду - паспортная выходная мощность DF-player - 3Вт. Это значит, что при питании 3.3 вольта с контроллера будет ток - 1 ампер. Насколько я помню макс на что рассчитан стаб платки ESP32 - 0.5А
То есть - в лучшем случает - плеер не вытянет нужную мощность, в худшем - сгорит стаб на плате и его нужно будет менять или просто выкидывать платку.

По диоду - да, часть плат выпускается с подключением диода на GPIO2, часть - на GPIO16. Переключатель в настройках - про это. Как влияет - не знаю, в исходниках ядра не лазил.

**Reply by chipm64** [reply]
Про потребление - верно, паспортная мощность на 5в. питания.
У меня нагрузка 8 ом, это уменьшит выходную мощность в 2 раза...до 1.5Вт (0,45А) 
Скорее всего запитаю от 5в+диод, меньше проблем  Управляющие выводы через резисторы 1к без преобразователя уровня.
Это самый простой вопрос в проекте.
Звуко-мелодии наверное можно добавлять свои? увеличивать номера имен файлов?
названия в web берутся из имени файлов или вбиты в исходник (может web-содержимое) ? тогда конечно не добавить

**Reply by vvip-68** [reply]
Имена файлов вбиты в прошивку на вкладке a_def_lang.h DFPlayer не умеет читать имена файлов и отдавать их контроллеру.
Я и резисторы обычно не подключаю. Резисторы были нужны при подключении к Arduino у которой 5-вольтовая логика. По крайней мере в инете схемы находились с резисторами. У этих же контроллеров как и у DFPlayer логика трехвольтовая - никаких резисторов и преобразователей думаю вообще не нужно. Я не ставил.
